src/app/collective-rewards/components/rewards/UpdateBackerRewardModal.tsx

```diff
@@ -0,0 +1,197 @@
+import {
+  useGetBackerRewardsForBuilder,
+  useSetBackerRewardsForBuilder,
+} from '@/app/collective-rewards/hooks/rewards/useBackerRewards'
+import { percentageToWei, weiToPercentage } from '@/app/collective-rewards/utils/rewards/backerRewards'
+import { useAlertContext } from '@/app/providers'
+import { AlertProps } from '@/components/Alert'
+import { Button } from '@/components/Button'
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormInputNumber,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/Form'
+import { LoadingSpinner } from '@/components/LoadingSpinner'
+import { Modal, ModalProps } from '@/components/Modal/Modal'
+import { Typography } from '@/components/Typography'
+import { Header } from '@/components/Typography/Header'
+import { config } from '@/config'
+import { TX_MESSAGES } from '@/shared/txMessages'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { waitForTransactionReceipt } from '@wagmi/core'
+import { FC } from 'react'
+import { useForm } from 'react-hook-form'
+import { GoRocket } from 'react-icons/go'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+import { z } from 'zod'
+
+const formSchema = z.object({
+  reward: z
+    .string()
+    .regex(
+      /^\d{1,3}(\.\d{1,18})?$/,
+      'Invalid format: should be up to 3 digits before decimal and up to 18 digits after decimal.',
+    )
+    .refine(
+      value => {
+        // Remove decimal point to check character count
+        const digitsOnly = value.replace('.', '')
+
+        // Ensure total character count without decimal point is <= 19
+        if (digitsOnly.length > 19) return false
+
+        const [whole, decimal] = value.split('.')
+
+        // Check that if there are 3 digits before the decimal, they must be "100" and there are no non-zero digits after 100.
+        return whole.length < 3 || (whole === '100' && BigInt(decimal) == 0n)
+      },
+      {
+        message:
+          'Value must be between 0.000_000_000_000_000_1 and 100.0, with total number of digits (excluding decimal point) less or equal to 19.',
+      },
+    ),
```

Comment:
One of the reasons for the form input to be a string is that as bigint it was allowing values over 100%
<img width="245" alt="Screenshot 2024-10-31 at 09 53 50" src="https://github.com/user-attachments/assets/45aafb0b-bfe7-4acb-a844-df5be03930e0">

Reply:
also, strangely enough, the conversion from wei to % is a bit easier this way